### PR TITLE
expose network info via /network_info

### PR DESCRIPTION
**Pull request by ailisp**
Mostly for getting peer address, so if loadtester knows one ip it can query for ip of other nodes in the network. Might also be useful for other applications. Currently this kind of information is available via push only (telemetry), this provides a pull way.
image

### CODE REVIEW

File: chain/client/src/info.rs [hunk 1]
```diff
@@ -94,6 +94,7 @@ impl InfoHelper {
               Green.bold().paint(format!("{:.2} bps {}", avg_bls, gas_used_per_sec(avg_gas_used))),
               Blue.bold().paint(format!("CPU: {:.0}%, Mem: {}", cpu_usage, pretty_bytes(memory * 1024)))
         );
+        println!("{:?}", network_info);
```
Comment (bowenwang1996) [inline chain/client/src/info.rs]: Remove?

File: chain/client/src/types.rs [hunk 2]
```diff
@@ -264,6 +266,29 @@ impl Message for Status {
...
 
+pub struct GetNetworkInfo {}
+
+impl Message for GetNetworkInfo {
+    type Result = Result<NetworkInfoResponse, String>;
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PeerInfo {
+    pub addr: Option<SocketAddr>,
```
Comment (bowenwang1996) [inline chain/client/src/types.rs]: Why is `addr` an option?

Comment (ailisp) [reply chain/client/src/types.rs]: The original network/src/types.rs FullPeerInfo.PeerInfo.addr is of type Option<SocketAddr> so I assumed it's possibly empty.

File: chain/client/src/info.rs (hunk 1 above)
Comment (ailisp) [reply chain/client/src/info.rs]: Yup

File: chain/client/src/types.rs [hunk 3]
```diff
@@ -264,6 +266,29 @@ impl Message for Status {
...
 }
 
+pub struct GetNetworkInfo {}
+
+impl Message for GetNetworkInfo {
+    type Result = Result<NetworkInfoResponse, String>;
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PeerInfo {
```
Comment (mfornet) [inline chain/client/src/types.rs]: With this one, there are three declaration of `PeerInfo` through the code. We should keep one declaration in `primitives` only.

Comment (ailisp) [reply chain/client/src/types.rs]: Good point! I didn't see the PeerInfo in primitives. If it's only in primitives and used in chain/netowork and chain/client that's great.

Comment (ilblackdragon) [reply chain/client/src/types.rs]: Should we only keep it in network? Ideally keep primitives as small as possible

Comment (ailisp) [reply chain/client/src/types.rs]: Sounds good. In a quick find, I do not see any usage of PeerInfo etc in primitives. Probably it has been moved to network & upper layer and definition in primitives are outdated. Will confirm then refactor

Comment (ailisp) [reply chain/client/src/types.rs]: Fixed, primitive/network.rs isn't used anywhere, removed. network/types.rs PeerInfo is now used

**Pull request merged**